source/arm9/math.c

```diff
@@ -0,0 +1,38 @@
+
+// SPDX-License-Identifier: Zlib
+// Copyright (C) 2024 Dominik Kurz
+#include <math.h>
+
+
+float hw_sqrtf(float x)
+{
+    union{f32 f; u32 i;}xu;
+    xu.f = x;
+    u64 mantissa = xu.i & ((1 << 23) - 1); // extracts mantissa via bitmask
```

Comment:
Some of your lines are too long. Try to keep them at most at 80 characters if possible, and definitely at less than 100 characters (for people that have two files open side by side).